File: donation-system-web/src/types/needGroup.ts
export interface NeedGroupResponse {
  id: string;
  name: string;
  active: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateNeedGroupRequest {
  name: string;
  active: boolean;
}

export interface UpdateNeedGroupRequest {
  name: string;
  active: boolean;
}

File: APISolidarityManager/APISolidarityManager/Entities/NeedGroup.cs
namespace APISolidarityManager.Models
{
    public class NeedGroup
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public bool Active { get; set; } = true;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public ICollection<NeedRule> NeedRules { get; set; } = new List<NeedRule>();
        public ICollection<ItemTemplate> ItemTemplates { get; set; } = new List<ItemTemplate>();
    }
}
